### DISCUSSION TITLE
xDrip sending double arrows down when sudden BG direction change.

### DISCUSSION THREAD

**Discussion by robertrub** [NightscoutFoundation/xDrip#3830] [Bug reports]
Hi

I thought that it was an AAPS bug but seems that it is an xDrip bug ;)

Please have a look at this issue (opened on AAPS) for the problem, the logs and screenshots and the remark from Milos.

https://github.com/nightscout/AndroidAPS/issues/3642

Thanks.

**Comment by Navid200** [comment]
Please have a look at this: https://github.com/NightscoutFoundation/xDrip/issues/414  

I will see if I can come up with a test to demonstrate that what xDrip broadcasts, as the trend arrow, is what it receives from G6.

**Reply by robertrub** [reply]
The first part of the discussion is not my case. The BGs are the same in xDrip and AAPS.

_I can make the following conclusions:_

_xDrip submits to Nightscout/AAPS/webservice the exact same values it receives from G6 or G7 as the value of delta and as the trend arrow slope._

I don't think so. As the delta and arrow direction/angle are correct on xDrip but on AAPS the arrows are double down (even when BGs are raising). AAPS calculates delta but uses the arrows that arrive from xDrip.

_xDrip calculates the delta between its current reading and its previous reading and uses that as the delta and trend arow slope that it shows on the xDrip main screen._

I think that xDrip should send the arrow direction it calculated to AAPS too.

I can send you xDrip logs if you want and tell me where to find them 😉

**Reply by robertrub** [reply]
One question : if I set ON _Send shown BG_ , will xDrip send the shown arrow too ?

**Reply by Navid200** [reply]
> I don't think so. As the delta and arrow direction/angle are correct on xDrip but on AAPS the arrows are double down (even when BGs are raising). AAPS calculates delta but uses the arrows that arrive from xDrip.   
  
Your conclusion should include a study of what Dexcom app shows as delta.  You cannot make that conclusion by just looking at what xDrip shows and what it broadcasts.  Those two are obviously different.  

The arrow is a calculated parameter.  It is not something that is measured by the filament under your skin.  The glucose is and xDrip broadcasts the glucose it receives from G6 as is.
Even AAPS should be able to calculate an arrow using those glucose values.

> One question : if I set ON Send shown BG , will xDrip send the shown arrow too ?  

I don't use AAPS.
But, there must be many who use G6 and xDrip and AAPS.  
There is nothing new in that respect.  The last time we changed anything in what xDrip broadcasts was a long time ago may be close to a year and it wasn't even related to G6.  It was related to G7.

I suggest asking in the facebook AAPS group.
